data/bigquery_sharded_data_pull.py

```diff
@@ -0,0 +1,75 @@
+"""
+Pull data >1Gb from GCP Bigquery using sharding functionality.
+Based on:
+    https://cloud.google.com/bigquery/docs/exporting-data#exporting_table_data
+    https://cloud.google.com/storage/docs/reference/libraries#using_the_client_library
+
+Author: Travis Couture
+Created: 03/17/2020
+Last Updated: 03/17/2020
+"""
+
+import datatable as dt
+import pandas as pd
+import os
+from h2oaicore.data import CustomData
+from h2oaicore.systemutils import config
+from google.cloud import bigquery
+from google.cloud import storage
+from functools import reduce
+
+
+# Please fill before usage
+# Note that this information is logged in Driverless AI logs.
+
+BUCKET_NAME = ''
+PROJECT = ''
+DATASET_ID = ''
+TABLE_ID = ''
+WILDCARD_NAME = ''
+LOCATION = ''
+DAI_DATA_PATH = ''
+
+
+class BqShardData(CustomData):
+
+    @staticmethod
+    def create_data(X: dt.Frame = None) -> dt.Frame:
+
+        sa_json = config.gcs_path_to_service_account_json
+
+        if sa_json.split('.')[-1] == 'json':
+            s_client = storage.Client.from_service_account_json(sa_json)
+            bq_client = bigquery.Client.from_service_account_json(sa_json)
+        else:
+            s_client = storage.Client()
+            bq_client = bigquery.Client()
+
+        destination_uri = 'gs://{}/{}'.format(BUCKET_NAME, WILDCARD_NAME)
+        dataset_ref = bq_client.dataset(DATASET_ID, project=PROJECT)
+        table_ref = dataset_ref.table(TABLE_ID)
+
+        extract_job = bq_client.extract_table(table_ref,
+                                              destination_uri,
+                                              location=LOCATION)
+
+        extract_job.result()
+
+        shard_count = extract_job.destination_uri_file_counts[0]
+        shard_name_prefix = WILDCARD_NAME.split('*')[0]
+        shard_filetype = WILDCARD_NAME.split('*')[1]
+        shard_list = [shard_name_prefix + str(i).zfill(12) + shard_filetype for i in range(0, shard_count)]
```

Comment:
q: is there a specific reason why we have to use `zfill` it seems weird.

Reply:
Is there not a way we can grab this via internal call to `GcsConnector` ls? would be cleaner and less likely to break on large queries. Like from here: https://github.com/h2oai/h2oai/blob/dev/h2oai/connectors/__init__.py#L500

Reply:
This was chosen for ease of implementation for a quick release. Since this should be permanently fixed in the next official release it shouldn't be an issue.

Reply:
yea I was just a bit worried about how robust this is... always better to let the service determine the naming and we just get the names, rather than the other way around... since this is just to unblock them is ok.

Reply:
let us table this as per discussion in slack to IF needed for more robust temporary fix.